qemu_v8.mk

```diff
@@ -0,0 +1,279 @@
+################################################################################
+# Following variables defines how the NS_USER (Non Secure User - Client
+# Application), NS_KERNEL (Non Secure Kernel), S_KERNEL (Secure Kernel) and
+# S_USER (Secure User - TA) are compiled
+################################################################################
+override COMPILE_NS_USER   := 64
+override COMPILE_NS_KERNEL := 64
+override COMPILE_S_USER    := 64
+override COMPILE_S_KERNEL  := 64
+
+-include common.mk
+
+################################################################################
+# Paths to git projects and various binaries
+################################################################################
+ARM_TF_PATH			?= $(ROOT)/arm-trusted-firmware
+
+EDK2_PATH			?= $(ROOT)/edk2
+EDK2_BIN			?= $(EDK2_PATH)/QEMU_EFI.fd
+
+QEMU_PATH			?= $(ROOT)/qemu
+
+SOC_TERM_PATH			?= $(ROOT)/soc_term
+STRACE_PATH			?= $(ROOT)/strace
+
+DEBUG = 1
+
+################################################################################
+# Targets
+################################################################################
+all: arm-tf qemu soc-term linux strace update_rootfs
+all-clean: arm-tf-clean busybox-clean linux-clean \
+	optee-os-clean optee-client-clean qemu-clean \
+	soc-term-clean check-clean strace-clean
+
+-include toolchain.mk
+
+################################################################################
+# ARM Trusted Firmware
+################################################################################
+# FIXME: Double check why Peter isn't using the bare metal compiler
+ARM_TF_EXPORTS ?= \
+	CFLAGS="-O0 -gdwarf-2" \
+	CROSS_COMPILE="$(CCACHE)$(AARCH64_CROSS_COMPILE)"
+
+ARM_TF_FLAGS ?= \
+	BL32=$(OPTEE_OS_BIN) \
+	BL33=$(EDK2_BIN) \
+	ARM_TSP_RAM_LOCATION=tdram \
+	PLAT=qemu \
+	DEBUG=0 \
+	LOG_LEVEL=50 \
+	ERROR_DEPRECATED=1 \
+	BL32_RAM_LOCATION=tdram \
+	SPD=opteed
+
+arm-tf: optee-os edk2
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) all fip
+
+arm-tf-clean:
+	$(ARM_TF_EXPORTS) $(MAKE) -C $(ARM_TF_PATH) $(ARM_TF_FLAGS) clean
+
+# FIXME: This is just too rough, we should build this just as we're doing for
```

Comment:
I think for now, this is good enough, but as soon as we find some time for it, we should as comment says build edk2 from source.